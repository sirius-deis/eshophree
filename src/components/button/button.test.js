import { render, screen, fireEvent } from '@testing-library/react';

import Button from './button';

describe('Button component', () => {
  it('should match snapshot with kind round', () => {
    const { container } = render(<Button kind='round'>Loading</Button>);
    expect(container).toMatchSnapshot();
  });
  it('should match snapshot with kind dropdown', () => {
    const { container } = render(<Button kind='dropdown'>Loading</Button>);
    expect(container).toMatchSnapshot();
  });
  it('should match snapshot with kind default', () => {
    const { container } = render(<Button>Loading</Button>);
    expect(container).toMatchSnapshot();
  });
  it('should fire event on dropdown type', () => {
    const fn = jest.fn();
    render(
      <Button kind={'dropdown'} onClickHandler={fn}>
        Loading
      </Button>,
    );
    fireEvent.click(screen.getByRole('button'));
    expect(fn).toHaveBeenCalled();
  });
});
