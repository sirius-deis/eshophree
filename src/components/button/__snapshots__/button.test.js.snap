// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot with kind default 1`] = `
<div>
  <button
    class="sc-bgqQcB sc-fFGjHI MyNZc gkWWIA"
  >
    Loading
  </button>
</div>
`;

exports[`Button component should match snapshot with kind dropdown 1`] = `
<div>
  <button
    class="sc-bgqQcB sc-ewnqHT MyNZc cTzBXJ"
    type="button"
  >
    Loading
  </button>
</div>
`;

exports[`Button component should match snapshot with kind round 1`] = `
<div>
  <button
    class="sc-bgqQcB sc-gTRrQi MyNZc cVPPIc"
    type="button"
  >
    Loading
  </button>
</div>
`;
